src/tools/perf/Makefile.am

```diff
@@ -33,7 +33,7 @@ ucx_perftest_LDADD    = \
 	$(abs_top_builddir)/src/uct/libuct.la \
 	$(abs_top_builddir)/src/ucp/libucp.la \
 	$(abs_top_builddir)/src/ucs/libucs.la \
-	lib/libucxperf.la
+	lib/libucxperf.la -lstdc++
```

Comment:
Following @Artemy-Mellanox comment here https://github.com/openucx/ucx/pull/10131#discussion_r1769582707 `-lstdc++` was missing to convert it to `std::vector`.
https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=89508&view=logs&jobId=e646a2ae-9139-5fa9-0995-85819d80e271&j=e646a2ae-9139-5fa9-0995-85819d80e271&t=40ba7047-ddc0-522c-15d2-d7fcc7934f9b
![image](https://github.com/user-attachments/assets/96f2e221-8d35-4967-8b59-387c4c7b45b4)

Reply:
> Following @Artemy-Mellanox comment here [#10131 (comment)](https://github.com/openucx/ucx/pull/10131#discussion_r1769582707) `-lstdc++` was missing to convert it to `std::vector`. https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=89508&view=logs&jobId=e646a2ae-9139-5fa9-0995-85819d80e271&j=e646a2ae-9139-5fa9-0995-85819d80e271&t=40ba7047-ddc0-522c-15d2-d7fcc7934f9b ![image](https://private-user-images.githubusercontent.com/80859800/375275982-96f2e221-8d35-4967-8b59-387c4c7b45b4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg1NDc1MDMsIm5iZiI6MTcyODU0NzIwMywicGF0aCI6Ii84MDg1OTgwMC8zNzUyNzU5ODItOTZmMmUyMjEtOGQzNS00OTY3LThiNTktMzg3YzRjN2I0NWI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDEwVDA4MDAwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTBjMGRiYzAyZDMxNWViMjQ0ZjNiYzc3MWVmYjY4ODBiYzVlZDdmZDQ3NmFmZWQ3MjllYzdmODQzMjY2YTA0YjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.byEF3MpNy0nEFNXYvkPV_Ei7ETdAK4YngGA98u0hwas)

we avoid libstdc++ on purpose
can we use malloc instead of vector?

Reply:
The reduction of stack usage here is by splitting to helper functions.
I'll revert it back to the static array of function pointers.

Also, @yosefe @Artemy-Mellanox https://github.com/openucx/ucx/pull/10131#issuecomment-2399612587 please let me know your preference so that we can proceed.

Reply:
if we avoid libstdc++ on purpose than need revert to static array

Reply:
done